### DISCUSSION TITLE
Uvicorn reload is always stuck at the except block and never runs the try part

### DISCUSSION THREAD

**Discussion by piyush-jhawar** [Kludex/uvicorn#2261] [Potential Issue]
from fastapi import FastAPI

app = FastAPI()

x = 1
while True:
    try:
         a = 1/x
         print("working fine")
         break
    except ZeroDivisionError:
         print("Division error")

First run the code with value of x as 1 and then change x to 0 and save, it will end up in except and change the value of x again to 1.

**Comment by Kludex** [comment]
Please provide an MRE.

**Reply by piyush-jhawar** [reply]
from fastapi import FastAPI


app = FastAPI()

x = 1
while True:
    try:
         a = 1/x
         print("working fine")
         break
    except ZeroDivisionError:
         print("Division error")


First run the code with value of x as 1 and then change x to 0 and save, it will end up in except and change the value of x again to 1.

**Reply by Kludex** [reply]
A minimal example doesn't involve the database... happy to help if you create an example I can run by just copy pasting.

**Reply by piyush-jhawar** [reply]
Extremely sorry for inconvenience caused, i have added an updated code. Let me know if this help.
